SPECS/golang/golang.spec

```diff
@@ -13,20 +15,21 @@
 %define __find_requires %{nil}
 Summary:        Go
 Name:           golang
-Version:        1.20.10
+Version:        1.21.6
 Release:        1%{?dist}
 License:        BSD-3-Clause
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Security
-URL:            https://golang.org
-Source0:        https://golang.org/dl/go%{version}.src.tar.gz
+URL:            https://github.com/microsoft/go
+Source0:        https://github.com/microsoft/go/releases/download/v%{version}-%{ms_go_revision}/go.%{ms_go_buildid}.src.tar.gz
 Source1:        https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz
-Source2:        https://dl.google.com/go/go%{bootstrap_compiler_version}.src.tar.gz
+Source2:        https://github.com/microsoft/go/releases/download/%{bootstrap_compiler_version}/go.20230802.5.src.tar.gz
 Patch0:         go14_bootstrap_aarch64.patch
 Obsoletes:      %{name} < %{version}
```

Comment:
I think we might want to remove the obsoletes..  If we are ultimately supporting the n and n-1 versions concurrently.  It might be that conflicts is more appropriate because we cannot have both installed at the same time....unless there is a way to do that.

Reply:
I'm not super familiar with the intricacies of `Obsoletes` vs. `Conflicts`, but I'm not sure if either is needed here. From what I can see, the package name has always been `golang`, so installing a specific version will either upgrade or downgrade an existing installed version. It seems pretty natural that a package will conflict with another version of itself so I guess it doesn't need to be stated in the spec. 😄 

`Obsoletes` seems to be more about handling one package(name) that no longer depends on another one, making it so that when installing the new version, the removed dependency will get cleaned up. Looking at all the `Obsoletes`es in https://src.fedoraproject.org/rpms/golang/blob/rawhide/f/golang.spec is interesting.

`Conflicts` seems like it's about conflicts between two different package(name)s. Although the intricacies [look complicated](https://docs.fedoraproject.org/en-US/packaging-guidelines/Conflicts) and it seems like it should be treated as a last resort.

Reply:
If the tooling properly allows installing them side-by-side and the package is installing the contents properly, then yeah, remove the Obsoletes and the conflicts.

Reply:
The Go tools work side by side, but this spec file doesn't define the package or install the files in a way that allows side-by-side installs.

Even so, the `Obsoletes: %{name} < %{version}` is not needed (nor any `Conflicts`), because this is already being treated as an upgrade. For example, https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/cmake/cmake.spec doesn't include `Obsoletes`, and it installs some non-side-by-side files like `/usr/bin/cmake`. When I run `tdnf install -y cmake-3.21.4-3.cm2.x86_64` (old) then run `tdnf install cmake`, it asks if I want to upgrade:

```
root [ / ]# tdnf install cmake
Loaded plugin: tdnfrepogpgcheck

Upgrading:
cmake              x86_64        3.21.4-10.cm2      mariner-official-base  46.54M      17.51M  
Total installed size:  46.54M
Total download size:  17.51M
Is this ok [y/N]:
```

As far as I can tell, it's theoretically possible to force dnf to install two versions on the same machine, but it's involved enough that I don't think anyone would do it (to any package, let alone golang) without either explicitly installing it to a different location or expecting some bad results.

Creating a set of RPM packages that work side by side seems doable to me, but not in the scope of this PR.

(I pushed a change to remove `Obsoletes:      %{name} < %{version}`.)